artemis-server/src/test/java/org/apache/activemq/artemis/core/config/BridgeConfigurationEncodingTest.java

```diff
@@ -0,0 +1,522 @@
+
+      // password
+      read = new byte[9];
+      data.readBytes(read, 0, 9);
+      assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 2, 0, 104, 0, 104}, read);
+
+      // connectionTtl
+      read = new byte[9];
+      data.readBytes(read, 0, 9);
+      assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 0, 0, 0, 0, 2}, read);
+
+      // maxRetryInterval
+      read = new byte[9];
+      data.readBytes(read, 0, 9);
+      assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 0, 0, 0, 0, 3}, read);
+
+      // minLargeMessageSize
+      read = new byte[5];
+      data.readBytes(read, 0, 5);
+      assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 5}, read);
+
+      // callTimeout
+      read = new byte[9];
+      data.readBytes(read, 0, 9);
+      assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 0, 0, 0, 0, 4}, read);
+
+      // concurrency
+      read = new byte[5];
+      data.readBytes(read, 0, 5);
+      assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 6}, read);
+
+      // configurationManaged
+      read = new byte[2];
+      data.readBytes(read, 0, 2);
+      assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0}, read);
+
+      // routingType
+      assertEquals(ComponentConfigurationRoutingType.ANYCAST.getType(), data.readByte());
+
+      if (transformer) {
+         // transformerClassName
+         read = new byte[9];
+         data.readBytes(read, 0, 9);
+         assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 2, 0, 106, 0, 106}, read);
+
+         if (properties) {
+            // number of transformer properties
+            read = new byte[4];
+            data.readBytes(read, 0, 4);
+            assertArrayEquals(new byte[] {0, 0, 0, 1}, read);
+
+            // transformer key name
+            read = new byte[9];
+            data.readBytes(read, 0, 9);
+            assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 2, 0, 107, 0, 107}, read);
+
+            // transformer key value
+            read = new byte[9];
+            data.readBytes(read, 0, 9);
+            assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 2, 0, 108, 0, 108}, read);
+         } else {
+            // transformer properties not present (indicated via 0 count)
+            read = new byte[4];
+            data.readBytes(read, 0, 4);
+            assertArrayEquals(new byte[] {0, 0, 0, 0}, read);
+         }
+      } else {
+         // transformer not present
+         assertEquals(DataConstants.NULL, data.readByte());
+      }
+
+      // staticConnector count
+      read = new byte[4];
+      data.readBytes(read, 0, 4);
+      assertArrayEquals(new byte[] {0, 0, 0, 1}, read);
+
+      // static connector
+      read = new byte[9];
+      data.readBytes(read, 0, 9);
+      assertArrayEquals(new byte[] {DataConstants.NOT_NULL, 0, 0, 0, 2, 0, 105, 0, 105}, read);
```

Comment:
Should probably also have test(s) without staticConnector, especially seeing as its size was incorrectly handled previously and is being fixed.

Reply:
Tests added.